Adding block-attributes-override filter (#24)
infinum/eightshift-docs: docs/guides/blocks-filer-block-attributes-override.md

---
id: blocks-filter-block-attributes-override
title: Filter Attributes Override
---

> This filter gives you and ability to hook you changes to blocks/wrapper attributes after it is registred in editor but before it is rendered in React.

Reviewer: ```suggestion
This filter gives you the ability to hook your changes to blocks/wrapper attributes after it is registered in the editor but before it is rendered in React.
```

Author: 👍

> This filter gives you and ability to hook you changes to blocks/wrapper attributes after it is registered in the block editor but before it is rendered in React.

Reviewer: ```suggestion
This filter gives you the ability to hook your changes to block/wrapper attributes after they are registered in the block editor but before they are rendered in React.
```

Author: 👍

> Usage of this method is vast. You can for example use it to provide differend blocks/wrapper defaults depending on the post type. For example you have a button block that you want to be default color red but on post type `post` you want that button to be black by default. Using this hook you can easily do this.

Reviewer: ```suggestion
The usage of this method is vast. You can, for example, use it to provide different blocks/wrapper defaults depending on the post type. For example, you have a button block that you want to have the color by default red, but on a post type `post`, you want that button to be black by default. Using this hook you can easily do this.
```

Author: 👍

> The usage of this method is vast. You can, for example, use it to provide different blocks/wrapper defaults depending on the post type. For example, you have a button block that you want to be default colour red but on post type `post`, you want that button to be black by default. Using this hook, you can easily do this.

Reviewer: ```suggestion
The usage of this method is vast. You can, for example, use it to provide different block/wrapper defaults depending on the post type. For example, you have a button block that you want to have the default color red, but on the post type `post`, you want that button to be black by default. Using this hook, you can easily do this.
```

Author: 👍

> > This method can only change the default values of the attributes. Once the value is saved in the database and it is different from the default it will stay in the detabase and all this changes will not apply to it.

Reviewer: ```suggestion
> This method can only change the default values of the attributes. Once the value is saved in the database and it is different from the default it will stay in the database and all the changes won't apply to it.
```

Author: 👍

> > This method can only change the default values of the attributes. Once the value is saved in the database and it is different from the default, it will stay in the database, and all these changes will not apply to it.

Reviewer: ```suggestion
> This method can only change the default values of the attributes. Once the value is saved in the database and it is different from the default, it will stay in the database, and all these changes won't apply to it.
```

Author: 👍

## Example

> If you havent allready create a new class src>blocks>class-blocks.php and extend `Eightshift_Libs\Blocks\Blocks` class from the lib.

Reviewer: ```suggestion
If you haven't already, create a new class in a file `src/blocks/class-blocks.php` and extend the `Eightshift_Libs\Blocks\Blocks` class from the lib.
```

Author: 👍

> Hot to extend a class [check here](extending-classes).

Reviewer: ```suggestion
How to extend a class [check here](extending-classes).
```

Author: 👍

> Then in the filter provide a callback method that contains overrides.

Reviewer: ```suggestion
Then in the filter provide a callback method that contains the overrides.
```

Author: 👍

```php
  /**
   * Register all the hooks
   *
   * @since 1.0.0
   *
   * @return void
   */
  public function register() {
    parent::register();

    add_filter( $this->config->get_config( static::BLOCK_ATTRIBUTES_FILTER_NAME ), [ $this, 'override_wrapper' ] );
  }

  /**
   * Override wrapper on all blocks.
   *
   * @param array $attr Array of blocks details.
   * @return array
   */

>   public function override_wrapper ( $attr ) : array {

Reviewer: ```suggestion
  public function override_wrapper ( array $attr ) : array {
```

Since we are adding a type hint on return type, might as well add it in the argument 🙂

Author: 👍
